workaround uClibc glob_pattern_p / __GLIBC__

Copyright (C) 2014 Curt Brune <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0

The uClibc header <features.h> defines the macro __GLIBC__, which
stricly speaking is false.  See the comment in the uClibc <features.h>
why they do this.

The poptconfig.c file tests for the presence of __GLIBC__ and
configures its globbing behaviour accordingly.  Unfortunately in the
case of uClibc it gets it wrong.

This patch simply inforces the correct behaviour when compiling with
ONIE's uClibc.
Summary:

diff --git a/poptconfig.c b/poptconfig.c
index 7840d0c..f4e115a 100644
--- a/poptconfig.c
+++ b/poptconfig.c
@@ -42,7 +42,7 @@ extern int glob_pattern_p (const char *__pattern, int __quote)
 /*@=declundef =exportheader =incondefs =protoparammatch =redecl =type @*/
 #endif	/* __LCLINT__ */
 
-#if !defined(__GLIBC__)
+#if !defined(__GLIBC__) || defined(__UCLIBC__)
 /* Return nonzero if PATTERN contains any metacharacters.
    Metacharacters can be quoted with backslashes if QUOTE is nonzero.  */
 static int
